### DISCUSSION TITLE
2 separate PWM channel into one CCT device?

### DISCUSSION THREAD

**Discussion by SandrRu** [Koenkk/zigbee2mqtt#5647] [Devices]
Is it possible to make a cct device from a dimer with two separate channels, using a custom converter for zigbee2mqtt ?

I have 2channel PWM controller with PTVO firmware.
I would like to control a light with variable temperature ( 2 kinds of LED strip 2700K and 5500k)

**Comment by merlin7g** [comment]
In the future, I would like to be able to control a full-color led strip with color temperature control. That is, having 5 PWM channels, control the RGB CCT tape from Home Assistant

**Reply by Koenkk** [reply]
A light with multiple channels can be defined like this; https://github.com/Koenkk/zigbee-herdsman-converters/blob/8842288db9a4e40f68636f11dc81a958fc9f937f/devices.js#L6622

**Reply by SandrRu** [reply]
This solution is for devices with firmware supporting multichannel color (CCT / RGBW) .
But in my case, the device has a firmware for 4 separate channels.
Is there a way to make multichannel control using a Herdsman-converter?

**Reply by Koenkk** [reply]
What's the difference between multichannel and 4 separate channels?

**Reply by SandrRu** [reply]
> What's the difference between multichannel and 4 separate channels?

i mean multicolor with cool2warm and rgb sliders....

**Reply by SandrRu** [reply]
Something like custom converter for multiendpoint device:

```
            if (
                msg.data.hasOwnProperty('presentValue')
            ) {
                result.color = {};

                if (model.meta && model.meta.multiEndpoint) {
                    const endpointName = model.hasOwnProperty('endpoint') ?
                        getKey(model.endpoint(msg.device), msg.endpoint.ID) : msg.endpoint.ID;
                }

                if (endpointName == 'l1') {
                    result.color.cool = precisionRound(msg.data['presentValue'] / 65535, 4);
                }
                if (endpointName == 'l2') {
                    result.color.warm = precisionRound(msg.data['presentValue'] / 65535, 4);
                }
            }
```
